### ISSUE TITLE
Change Request: Disallow multiple configuration comments for the same rule

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#18132]
### ESLint version

v9.0.0-beta.0

### What problem do you want to solve?

Example:

```js
/* eslint semi: ["error", "never"] */

foo

/* eslint semi: ["error", "always"] */

foo;
```

```
  3:4  error  Missing semicolon  semi
```

Allowing multiple configuration comments for the same rule can lead to the conclusion that it's possible to have different configurations for different parts of the file, while actually the last configuration applies to the whole file.

It's also confusing, now when comments with just severity retain previously configured options (https://github.com/eslint/eslint/issues/17381), what should be the final configuration in cases like this:

```js
/* eslint semi: ["error", "never"] */

/* eslint semi: ["warn"] */

// should the final configuration be semi: ["warn", "never"] ?
```

### What do you think is the correct solution?

Disallow multiple configuration comments for the same rule.

The first comment should apply, while the others should be reported as lint errors.

### Participation

- [X] I am willing to submit a pull request for this change.

### Additional comments

This makes two breaking changes:

1. Reporting multiple configuration comments as lint errors is a breaking change.
2. The first comment will apply, while it used to be the last comment. This could be avoided by applying the last comment and reporting previous ones, but I think it makes more sense to report those that appear later.

**Comment by mdjermanovic** [comment]
I'm working on this.
